src/NServiceBus.Core/Serializers/MessageDeserializerResolver.cs

```diff
@@ -12,14 +12,17 @@ class MessageDeserializerResolver
 
         public MessageDeserializerResolver(IEnumerable<IMessageSerializer> messageSerializers, Type defaultSerializerType)
         {
-            Guard.AgainstNull("defaultSerializerType", defaultSerializerType);
-
             serializersMap = messageSerializers.ToDictionary(key => key.ContentType, value => value);
             defaultSerializer = serializersMap.Values.Single(serializer => serializer.GetType() == defaultSerializerType);
         }
 
         public IMessageSerializer Resolve(string contentType)
         {
+            if (contentType == null)
```

Comment:
Here is what I think.
The `MessageDeserializerResolver.Resolve` should not be returning the default deserializer if a null is passed to it (since a content type of `null`, really means, I have no idea what it is), instead the consumer of the api needs to make an explicit decision to attempt to perform the deserialization against the default deserializer. So no ownership on  `MessageDeserializerResolver` to make that guess work.